packages/braintree-integration/src/braintree-paypal/braintree-paypal-payment-strategy.ts

```diff
@@ -138,12 +178,17 @@ export default class BraintreePaypalPaymentStrategy implements PaymentStrategy {
         } = this.paymentMethod;
         const { methodId, paymentData = {} } = payment;
 
-        if (nonce) {
+        const token = this.braintreeTokenizePayload?.nonce || nonce;
+
+        if (token) {
             const state = await this.paymentIntegrationService.loadPaymentMethod(methodId);
 
             this.paymentMethod = state.getPaymentMethod(methodId);
 
-            return Promise.resolve({ ...payment, paymentData: this.formattedPayload(nonce) });
+            return Promise.resolve({
+                ...payment,
+                paymentData: this.formattedPayload(token),
+            });
```

Comment:
It is not necessary to make `Promise.resolve` here

```suggestion
            return {
                ...payment,
                paymentData: this.formattedPayload(token),
            };
```